### PR TITLE
Upgrade kubernetes version & move lifecycle attr. to e2e tests

### PR DESCRIPTION
**Pull request by inteon**
This PR contains the code generated by the https://github.com/cert-manager/release/pull/97, https://github.com/cert-manager/release/pull/102 and https://github.com/cert-manager/release/pull/98 PRs in the release repo.
Generation was done as follows:
```shell
$ gh --repo https://github.com/cert-manager/release run download
$ git apply prow-config.patch
```

Signed-off-by: Tim Ramlot <42113979+inteon@users.noreply.github.com>

### CODE REVIEW

**Comment by maelvls** [comment]
Thank you for adding 1.25 to the Prow jobs.

/lgtm

File: config/jobs/cert-manager/cert-manager/master/cert-manager-master.yaml [hunk 1]
```diff
@@ -30,13 +30,6 @@ presubmits:
           requests:
             cpu: 2000m
             memory: 4Gi
-        lifecycle:
-          preStop:
-            exec:
-              command:
-              - /bin/sh
-              - -c
-              - make kind-logs
```
[the remark this answers is not in the document]
Comment (inteon) [reply config/jobs/cert-manager/cert-manager/master/cert-manager-master.yaml:39]: fixed in https://github.com/cert-manager/release/pull/102

**Review by jahrlin — APPROVED** [review]
/lgtm
/approve

**Comment by inteon** [comment]
@jahrlin This caused the cert-manager to fail 😞. However, it is not clear to me how we should test adding 1.25 before making this change. In this cert-manager PR, I'm adding support for 1.25 kind: https://github.com/cert-manager/cert-manager/pull/5459.

**Pull request merged**